### ISSUE TITLE
Extension system

### ISSUE THREAD

**Issue by alvarlagerlof** [wilfred-dev/wilfred#34]
**Is your feature request related to a problem? Please describe.**
Some things like my Overviewer map is thightly coupled to my server, yet does not fit as a general feature in wilfred.

**Describe the solution you'd like**
I'd like to be able to develop python extensions that use some kind of API that Wilfred exposes. The API could for example expose:
- list of servers
- server state
- server start/stop
- server path
- sever console (for announcing rendering)

With only these ones, you'd be able to make a plugin for an Overviewer map.

**Describe alternatives you've considered**
Integrating support for Overviewer directly in Wilfred.

**Additional context**
None.

**Comment by VilhelmPrytz** [comment]
This is a really interesting feature that should definitely be implemented. It's also something that would probably take some time to implement.

What types of extensions are you thinking about? Like, extensions that closely integrate to the config (like linking BungeeCord servers together) or extensions that add new functionality and commands to the interface?

**Comment by alvarlagerlof** [comment]
I think that both would be needed. Otherwise there would need to be config files for extensions and that seems annoying.

**Comment by vilhelmprytz** [comment]
This has now been queued to be implemented after #60 for the `v0.8.0` release, see https://github.com/wilfred-dev/wilfred/projects/1

**Comment by vilhelmprytz** [comment]
Postponed to `v0.9.0`.
